packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -872,11 +871,18 @@ abstract class DatabaseInstanceSource extends DatabaseInstanceNew implements IDa
       }
     }
 
+    // if engine version is pinned and autoMinorVersionUpgrade is not passed in props, we set this property to false
+    let autoMinorVersionUpgrade = props.autoMinorVersionUpgrade;
```

Comment:
Same comments here:

1. Get rid of the `autoMinorVersionUpgrade` variable, and do this work inline.
2. Check `props.engine.engineVersion?.fullVersion`.